### ISSUE TITLE
twisted.protocols.amp.Command should have methods for argument serialization and parsing

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2667]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2667|
|Type|enhancement|
|Created|2007-05-20 18:18:29Z|

This will make it easier to write unit tests.

<details><summary>Searchable metadata</summary>

```
trac-id__2667 2667
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1179685109000000 1179685109000000
changetime__1179934645000000 1179934645000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix set status to `assigned`|
|-|-|

See [#2657](https://github.com/twisted/twisted/issues/2657) for where the work for this fix is happening.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix set status to `closed`|
|-|-|

(In [20391]) Merge amp-factoring-2657+2667.

Author: radix
Reviewer: therve
Fixes [[#2657](https://github.com/twisted/twisted/issues/2657)](https://github.com/twisted/twisted/issues/2657)
Fixes [[#2667](https://github.com/twisted/twisted/issues/2667)](https://github.com/twisted/twisted/issues/2667)
Fixes [[#2656](https://github.com/twisted/twisted/issues/2656)](https://github.com/twisted/twisted/issues/2656)
Fixes [[#2664](https://github.com/twisted/twisted/issues/2664)](https://github.com/twisted/twisted/issues/2664)

It is now possible to use public methods on Command
objects to serialize and parse arguments and responses
based on that Commandn's arguments and response schemas.
It was already possible to serialize responses via
the Command in this way; analogs for the other three cases
were added. (#2657 & #2667).

Some docstrings in nearby code were alse fixed in this branch.
(#2656 & #2664).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
